### PR TITLE
[DOCS] Add collapsible sections to 8.0 breaking changes [Part 3]

### PR DESCRIPTION
**Pull request by jrodewig**
Adds collapsible sections to the 8.0 breaking changes.

To make reviews easier and avoid introducing errors, I've
tried to limit my changes to formatting, with the exception
of rewording heading as sentences.

I'm also splitting this effort over several PRs to make reviews
manageable.

Relates to #55629 and #53229

### Preview
http://elasticsearch_56297.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/breaking-changes-8.0.html

### CODE REVIEW

**Review by debadair — APPROVED** [review]
LGTM

File: docs/reference/migration/migrate_8_0/http.asciidoc [hunk 1]
```diff
@@ -8,16 +8,31 @@
...
+result in an error on startup.
+====
 
+.The `es.rest.url_plus_as_space` system property has been removed.
+[%collapsible]
+====
+*Details* +
 Starting in version 7.4, a `+` in a URL will be encoded as `%2B` by all REST API functionality. Prior versions handled a `+` as a single space.
 If your application requires handling `+` as a single space you can return to the old behaviour by setting the system property
 `es.rest.url_plus_as_space` to `true`. Note that this behaviour is deprecated and setting this system property to `true` will cease
```
Comment (debadair) [inline docs/reference/migration/migrate_8_0/http.asciidoc:30]: This seems contradictory with the impact statement.

**Review by debadair — APPROVED** [review]
One bit that looks like it needs to be clarified, otherwise LGTM.

File: docs/reference/migration/migrate_8_0/http.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/migration/migrate_8_0/http.asciidoc]: Yeah. The following sentence makes it clear(er), and I did test to confirm that this property does get rejected. I reworded the original sentence with 001b32817f1d7a505bd2baa4b5c00cb4b1e03cd1.

**Pull request merged**